prefect_dbt/cloud/jobs.py

```diff
@@ -318,9 +325,167 @@ def my_flow():
+    return trigger_job_run_options_override
+
+
+@flow(
+    name="Retry subset of dbt Cloud job run and wait for completion",
+    description=(
+        "Retries a subset of dbt Cloud job run, filtered by select statuses, "
+        "and waits for the triggered retry to complete."
+    ),
+)
+async def retry_dbt_cloud_job_run_subset_and_wait_for_completion(
+    dbt_cloud_credentials: DbtCloudCredentials,
+    run_id: int,
+    trigger_job_run_options: Optional[TriggerJobRunOptions] = None,
+    max_wait_seconds: int = 900,
+    poll_frequency_seconds: int = 10,
+    retry_status_filters: List[str] = ["error", "fail"],
+    retry_downstream_nodes: bool = True,
+) -> Dict:
+    """
+    Flow that retrys a subset of dbt Cloud job run, filtered by select statuses,
+    and waits for the triggered retry to complete.
+
+    Args:
+        dbt_cloud_credentials: Credentials for authenticating with dbt Cloud.
+        trigger_job_run_options: An optional TriggerJobRunOptions instance to
+            specify overrides for the triggered job run.
+        max_wait_seconds: Maximum number of seconds to wait for job to complete
+        poll_frequency_seconds: Number of seconds to wait in between checks for
+            run completion.
+        run_id: The ID of the job run to retry.
+        retry_status_filters: A list of statuses to filter the models by.
+        retry_downstream_nodes: Whether to also retry nodes downstream of the filtered models.
+
+    Raises:
+        ValueError: If `trigger_job_run_options.steps_override` is set by the user.
+
+    Returns:
+        The run data returned by the dbt Cloud administrative API.
+
+    Examples:
+        Retry a subset of models in a dbt Cloud job run and wait for completion:
+        ```python
+        from prefect import flow
+
+        from prefect_dbt.cloud import DbtCloudCredentials
+        from prefect_dbt.cloud.jobs import retry_dbt_cloud_job_run_subset_and_wait_for_completion
+
+        @flow
+        def retry_dbt_cloud_job_run_subset_and_wait_for_completion_flow():
+            credentials = DbtCloudCredentials.load("MY_BLOCK_NAME")
+            retry_dbt_cloud_job_run_subset_and_wait_for_completion(
+                dbt_cloud_credentials=credentials,
+                run_id=88640123,
+                retry_status_filters=("error", "fail", "warn"),
+                retry_downstream_nodes=True,
+            )
+
+        retry_dbt_cloud_job_run_subset_and_wait_for_completion_flow()
+        ```
+    """  # noqa
+    if trigger_job_run_options and trigger_job_run_options.steps_override is not None:
+        raise ValueError(
+            "Do not set `steps_override` in `trigger_job_run_options` "
+            "because this flow will automatically set it"
+        )
+
+    run_artifact_future = await get_dbt_cloud_run_artifact.submit(
+        dbt_cloud_credentials=dbt_cloud_credentials,
+        run_id=run_id,
+        path="run_results.json",
+    )
```

Comment:
Without a `step` parameter here, this will return the `run_results.json` file from the last step in the run, which may not be the failed step if I've also checked "Generate Docs" when setting up my job.  That doc generation step will run regardless of status for prior steps.  In that case, you won't actually get the nodes that failed from the prior step.

![image](https://user-images.githubusercontent.com/10198967/195834658-158cd3a0-4c24-487c-b86a-df57e661fbf0.png)